swift_package/BrainFlow/Sources/BoardShim.swift

```diff
@@ -0,0 +1,606 @@
+     */
+    func releaseSession () throws {
+        try? BoardShim.logMessage(.LEVEL_INFO, "release session")
+        var jsonBFParams = self.jsonBrainFlowInputParams
+        let errorCode = release_session (boardId.rawValue, &jsonBFParams)
+        try checkErrorCode("failed to release session", errorCode)
+    }
+    
+    /**
+     * get number of packages in ringbuffer
+     */
+    func getBoardDataCount (_ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> Int32 {
+        var dataCount: Int32 = 0
+        var jsonBFParams = self.jsonBrainFlowInputParams
+        let errorCode = get_board_data_count (preset.rawValue, &dataCount, boardId.rawValue, &jsonBFParams)
+        try checkErrorCode("failed to get board data count", errorCode)
+        
+        return dataCount
+    }
+
+    /**
+     * insert marker to data stream
+     */
+    func insertMarker (value: Double, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws {
+        var jsonBFParams = self.jsonBrainFlowInputParams
+        let errorCode = insert_marker (value, preset.rawValue, boardId.rawValue, &jsonBFParams)
+        try checkErrorCode("Error in insert_marker", errorCode)
+    }
+    
+    /**
+     * check session status
+     */
+    func isPrepared () throws -> Bool  {
+        var intPrepared: Int32 = 0
+        var jsonBFParams = self.jsonBrainFlowInputParams
+        let errorCode = is_prepared (&intPrepared, boardId.rawValue, &jsonBFParams)
+        try checkErrorCode("failed to check session", errorCode)
+        
+        return NSNumber(value: intPrepared).boolValue
+    }
+
+    /**
+     * get latest collected data, can return less than "num_samples", doesnt flush
+     * it from ringbuffer
+     */
+    func getCurrentBoardData (_ numSamples: Int32, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> [[Double]]
+    {
+        var numRows: Int32 = 0
+        var currentSize: Int32 = 0
+        var jsonBFParams = self.jsonBrainFlowInputParams
+
+        do {
+            numRows = try BoardShim.getNumRows (getBoardId()) }
+        catch {
+            throw error
+        }
+            
+        var buffer = [Double](repeating: 0.0, count: Int(numSamples * numRows))
+        let errorCode = get_current_board_data (numSamples, preset.rawValue, &buffer, &currentSize,
+                                                boardId.rawValue, &jsonBFParams)
+        try checkErrorCode("Error in get_current_board_data", errorCode)
+        
+        return buffer.matrix2D(rowLength: Int(numSamples))
+    }
+
+    /**
+     * get all data from the default ringbuffer and flush it
+     */
+    func getDefData () throws -> [[Double]] {
+        let size = try getBoardDataCount()
+        guard size > 0 else {
+            return [[Double]]()
+        }
+        
+        return try getBoardData(size)
+    }
+    
+    /**
+     * get all data from the auxiliary ringbuffer and flush it
+     */
+    func getAuxData () throws -> [[Double]] {
```

Comment:
same here and below

Reply:
Removed. Should I also remove getAncData()?

Reply:
Yes